File: app/src/main/java/com/angus/gaming/blackredsolitaire/ToPlayFragment.java
package com.angus.gaming.blackredsolitaire;

import android.app.Fragment;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * A simple {@link Fragment} subclass.
 */
public class ToPlayFragment extends Fragment {

    @BindView(R.id.hint_textview)
    TextView hintTextview;

    @BindView(R.id.hint_imageview)
    ImageView hintImageview;

    @BindView(R.id.previousButton)
    Button previousButton;

    @BindView(R.id.nextButton)
    Button nextButton;

    @BindView(R.id.toPlayFragmentLayout)
    RelativeLayout toPlayFragmentLayout;

    private int hintNumber = 1;

    public ToPlayFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_to_play, container, false);
        ButterKnife.bind(this, view);

        // Inflate the layout for this fragment
        return view;
    }

    @OnClick(R.id.nextButton)
    void onNextButtonClick(View v){
        if(hintNumber!=9) {
            hintNumber++;
            onButtonClick();
        }
    }

    @OnClick(R.id.previousButton)
    void onPreviousButton(){
        if(hintNumber!=1) {
            hintNumber--;
            onButtonClick();
        }
    }

    private void onButtonClick(){
        switch (hintNumber) {
            case 1:
                hintTextview.setText(R.string.hintOne);
                previousButton.setVisibility(View.INVISIBLE);
                toPlayFragmentLayout.setBackgroundResource(R.drawable.bottom_row_shape);
                hintImageview.setImageDrawable(getResources().getDrawable(
                        R.drawable.hint_one,null));
                break;
            case 2:
                hintTextview.setText(R.string.hintTwo);
                previousButton.setVisibility(View.VISIBLE);
                toPlayFragmentLayout.setBackgroundResource(R.drawable.locked_level_shape);
                hintImageview.setImageDrawable(
                        getResources().getDrawable(R.drawable.hint_two, null));
                break;
            case 3:
                hintTextview.setText(R.string.hintThree);
                toPlayFragmentLayout.setBackgroundResource(R.drawable.bottom_row_shape);
                hintImageview.setImageDrawable(getResources().getDrawable(
                        R.drawable.hint_three, null));
                break;
            case 4:
                hintTextview.setText(R.string.hintFour);
                toPlayFragmentLayout.setBackgroundResource(R.drawable.bottom_row_shape);
                hintImageview.setImageDrawable(getResources().getDrawable(
                        R.drawable.hint_four, null));
                break;
            case 5:
                hintTextview.setText(R.string.hintFive);
                toPlayFragmentLayout.setBackgroundResource(R.drawable.current_level_shape);
                hintImageview.setImageDrawable(getResources().getDrawable(
                        R.drawable.hint_five, null));
                break;
            case 6:
                hintTextview.setText(R.string.hintSix);
                toPlayFragmentLayout.setBackgroundResource(R.drawable.face_number_shape);
                hintImageview.setImageDrawable(getResources().getDrawable(
                        R.drawable.hint_six));
                break;
            case 7:
                hintTextview.setText(R.string.hintSeven);
                toPlayFragmentLayout.setBackgroundResource(R.drawable.face_number_shape);
                hintImageview.setImageDrawable(getResources().getDrawable(
                        R.drawable.hint_seven, null));
                break;
            case 8:
                hintTextview.setText(R.string.hintEight);
                nextButton.setVisibility(View.VISIBLE);
                hintImageview.setImageDrawable(getResources().getDrawable(
                        R.drawable.hint_eight, null));
                break;
            case 9:
                hintTextview.setText(R.string.hintNine);
                nextButton.setVisibility(View.GONE);
                hintImageview.setImageDrawable(getResources().getDrawable(
                        R.drawable.hint_nine, null));
                break;
        }
    }
}

File: app/src/main/java/gms/angus/angussoli/view/ToPlayFragment.kt
package gms.angus.angussoli.view

import android.os.Bundle
import android.util.TypedValue
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import gms.angus.angussoli.R
import gms.angus.angussoli.databinding.FragmentToPlayBinding

class ToPlayFragment : Fragment() {
    var hintNumber = 1

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {

        val binding = FragmentToPlayBinding.inflate(inflater, container, false)
        binding.nextButton.setOnClickListener {
            if (hintNumber != 9) {
                hintNumber++
                onButtonClick(binding)
            }
        }
        binding.previousButton.setOnClickListener {
            if (hintNumber != 1) {
                hintNumber--
                onButtonClick(binding)
            }
        }
        return binding.root
    }

    private fun onButtonClick(binding: FragmentToPlayBinding) {
        val hintTextview = binding.hintTextview
        val titleTextview = binding.titleTextview
        val hintImageview = binding.hintImageview
        val previousButton = binding.previousButton
        val nextButton = binding.nextButton
        val toPlayFragmentLayout = binding.root
        when (hintNumber) {
            1 -> {
                hintTextview.setText(R.string.hintOne)
                titleTextview.setText(R.string.hintOneTitle)
                previousButton.visibility = View.INVISIBLE
                toPlayFragmentLayout.setBackgroundResource(R.drawable.zone_deck)
                hintImageview.setImageDrawable(
                    resources.getDrawable(
                        R.drawable.hint_one, null
                    )
                )
            }
            2 -> {
                hintTextview.setText(R.string.hintTwo)
                titleTextview.setText(R.string.hintTwoTitle)
                previousButton.visibility = View.VISIBLE
                toPlayFragmentLayout.setBackgroundResource(R.drawable.zone_locked)
                hintImageview.setImageDrawable(
                    resources.getDrawable(R.drawable.hint_two, null)
                )
            }
            3 -> {
                hintTextview.setText(R.string.hintThree)
                titleTextview.setText(R.string.hintThreeTitle)
                TypedValue().let{
                    activity?.theme?.resolveAttribute(R.attr.greyTextColor, it, true)
                    it.data
                }.let {
                    hintTextview.setTextColor(it)
                    titleTextview.setTextColor(it)
                }
                toPlayFragmentLayout.setBackgroundResource(R.drawable.zone_deck)
                hintImageview.setImageDrawable(
                    resources.getDrawable(
                        R.drawable.hint_three, null
                    )
                )
            }
            4 -> {
                hintTextview.setText(R.string.hintFour)
                titleTextview.setText(R.string.hintFourTitle)
                TypedValue().let{
                    activity?.theme?.resolveAttribute(R.attr.negativeTextColor, it, true)
                    it.data
                }.let {
                    hintTextview.setTextColor(it)
                    titleTextview.setTextColor(it)
                }
                toPlayFragmentLayout.setBackgroundColor(TypedValue().let{
                    activity?.theme?.resolveAttribute(R.attr.backgroundColor, it, true)
                    it.data
                })
                hintImageview.setImageDrawable(
                    resources.getDrawable(
                        R.drawable.hint_four, null
                    )
                )
            }
            5 -> {
                hintTextview.setText(R.string.hintFive)
                titleTextview.setText(R.string.hintFiveTitle)
                TypedValue().let{
                    activity?.theme?.resolveAttribute(R.attr.positiveTextColor, it, true)
                    it.data
                }.let{
                    hintTextview.setTextColor(it)
                    titleTextview.setTextColor(it)
                }
                toPlayFragmentLayout.setBackgroundResource(R.drawable.zone_current)
                hintImageview.setImageDrawable(
                    resources.getDrawable(
                        R.drawable.hint_five, null
                    )
                )
            }
            6 -> {
                hintTextview.setText(R.string.hintSix)
                titleTextview.setText(R.string.hintSixTitle)
                TypedValue().let{
                    activity?.theme?.resolveAttribute(R.attr.positiveTextColor, it, true)
                    it.data
                }.let {
                    hintTextview.setTextColor(it)
                    titleTextview.setTextColor(it)
                }
                toPlayFragmentLayout.setBackgroundResource(R.drawable.zone_cleared)
                hintImageview.setImageDrawable(
                    resources.getDrawable(
                        R.drawable.hint_six, null
                    )
                )
            }
            7 -> {
                hintTextview.setText(R.string.hintSeven)
                toPlayFragmentLayout.setBackgroundResource(R.drawable.zone_cleared)
                titleTextview.setText(R.string.hintSevenTitle)
                TypedValue().let{
                    activity?.theme?.resolveAttribute(R.attr.positiveTextColor, it, true)
                    it.data
                }.let{
                    hintTextview.setTextColor(it)
                    titleTextview.setTextColor(it)
                }
                hintImageview.setImageDrawable(
                    resources.getDrawable(
                        R.drawable.hint_seven, null
                    )
                )
            }
            8 -> {
                hintTextview.setText(R.string.hintEight)
                toPlayFragmentLayout.setBackgroundResource(R.drawable.zone_broken)
                titleTextview.setText(R.string.hintEightTitle)
                TypedValue().let{
                    activity?.theme?.resolveAttribute(R.attr.lockedZoneBackgroundColor, it, true)
                    it.data
                }.let{
                    hintTextview.setTextColor(it)
                    titleTextview.setTextColor(it)
                }
                nextButton.visibility = View.VISIBLE
                hintImageview.setImageDrawable(
                    resources.getDrawable(
                        R.drawable.hint_eight, null
                    )
                )
            }
            9 -> {
                hintTextview.setText(R.string.hintNine)
                titleTextview.setText(R.string.hintNineTitle)
                nextButton.visibility = View.GONE
                hintImageview.setImageDrawable(
                    resources.getDrawable(
                        R.drawable.hint_nine, null
                    )
                )
            }
        }
    }
}